x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/embeddings/AzureOpenAiEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,339 @@
+        return switch (context) {
+            case REQUEST -> fromRequestMap(map);
+            case PERSISTENT -> fromPersistentMap(map);
+        };
+    }
+
+    private static AzureOpenAiEmbeddingsServiceSettings fromPersistentMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        var settings = fromMap(map, validationException);
+
+        Boolean dimensionsSetByUser = removeAsType(map, DIMENSIONS_SET_BY_USER, Boolean.class);
+        if (dimensionsSetByUser == null && map.containsKey(DIMENSIONS)) {
+            validationException.addValidationError(
+                ServiceUtils.missingSettingErrorMsg(DIMENSIONS_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+            );
+        }
+
+        Boolean encodingFormatSetByUser = removeAsType(map, ENCODING_FORMAT_SET_BY_USER, Boolean.class);
+        if (encodingFormatSetByUser == null && map.containsKey(ENCODING_FORMAT)) {
+            validationException.addValidationError(
+                ServiceUtils.missingSettingErrorMsg(ENCODING_FORMAT_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+            );
+        }
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new AzureOpenAiEmbeddingsServiceSettings(
+            settings,
+            Boolean.TRUE.equals(dimensionsSetByUser),
+            Boolean.TRUE.equals(encodingFormatSetByUser)
+        );
+    }
+
+    private static AzureOpenAiEmbeddingsServiceSettings fromRequestMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        var commonFields = fromMap(map, validationException);
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new AzureOpenAiEmbeddingsServiceSettings(commonFields, commonFields.dimensions != null, commonFields.encodingFormat != null);
+    }
+
+    private static CommonFields fromMap(Map<String, Object> map, ValidationException validationException) {
+        String resourceName = extractRequiredString(map, RESOURCE_NAME, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        String deploymentId = extractRequiredString(map, DEPLOYMENT_ID, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        String apiVersion = extractRequiredString(map, API_VERSION, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        Integer dims = removeAsType(map, DIMENSIONS, Integer.class);
+        String encodingFormat = extractOptionalString(map, ENCODING_FORMAT, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        Integer maxTokens = removeAsType(map, MAX_INPUT_TOKENS, Integer.class);
+
+        if (dims != null) {
+            Boolean hasUserSetDimensions = extractOptionalBoolean(
```

Comment:
Add the `hasUserSetX` values as members of CommonFields and allow the calling function to make judgements about the validation